sample/src/main/java/co/nimblehq/sample/ui/login/LoginViewModel.kt

```diff
@@ -3,19 +3,33 @@ package co.nimblehq.sample.ui.login
 import androidx.lifecycle.LiveData
 import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import co.nimblehq.sample.AuthenService
+import co.nimblehq.survey.sdk.SurveyApi
+import co.nimblehq.survey.sdk.request.LoginRequest
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
 
-class LoginViewModel() : ViewModel() {
-
-    private val _loginForm = MutableLiveData<LoginFormState>()
-    val loginFormState: LiveData<LoginFormState> = _loginForm
-
+class LoginViewModel : ViewModel() {
     private val _loginResult = MutableLiveData<LoginResult>()
     val loginResult: LiveData<LoginResult> = _loginResult
+    val authenService = SurveyApi.getInstance().buildService<AuthenService>()
 
-    fun login(username: String, password: String) {
-        // can be launched in a separate asynchronous job
-        _loginResult.value =
-            LoginResult(success = LoggedInUserView("test", "token"))
-    }
+    fun login() {
+        val request = LoginRequest(grantType = "password", email = "dev@nimblehq.co", "12345678")
```

Comment:
when using any hard-coded values, please add a comment on your intention here, will they be replaced? We avoid using hard-coded values in the code, any exception should be explained 🙏

Reply:
@luongvo Okay, let me put more comment on it.